astropy/io/ascii/ecsv.py

```diff
@@ -210,3 +227,43 @@ class Ecsv(basic.Basic):
 
     header_class = EcsvHeader
     outputter_class = EcsvOutputter
+
+    def update_table_data(self, table):
+        """
+        Update table columns in place if mixin columns are present.
+
+        This is a hook to allow updating the table columns after name
+        filtering but before setting up to write the data.  This is currently
+        only used by ECSV and is otherwise just a pass-through.
+
+        Parameters
+        ----------
+        table : `astropy.table.Table`
+            Input table for writing
+
+        Returns
+        -------
+        table : `astropy.table.Table`
+            Output table for writing
+        """
+        @contextlib.contextmanager
```

Comment:
On the dog walk I thought of a better way -- a new TimeInfo attribute `serialize_as` which is a dict that indicates how various serialization engines should behave.  So something like:
```
class TimeInfo(...):
     serialize_formatted_values = {'fits': False, 'ecsv': True, 'hdf5': False, 'yaml': False}
```
In theory this could be put into `MixinInfo`.  Not sure if that should be done before there is a use case, but probably `SkyCoord` will want this.  For `SkyCoord` this would choose between the (new) formatted output vs. the decimal in the default preferred units.